pkg/stream/stream.go

```diff
@@ -0,0 +1,158 @@
+// Copyright © 2018 Infostellar, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stream
+
+import (
+	"context"
+	"io"
+	"log"
+	"sync/atomic"
+
+	"google.golang.org/grpc"
+	stellarstation "github.com/infostellarinc/go-stellarstation/api/v1"
+
+	"github.com/infostellarinc/stellarcli/pkg/apiclient"
+)
+
+const (
+	OPEN   uint32 = 0
+	CLOSED uint32 = 1
+)
+
+type SatelliteStream interface {
+	Start() error
+	Send(payload []byte) error
+
+	io.Closer
+}
+
+type satelliteStream struct {
+	satelliteId string
+	stream      stellarstation.StellarStationService_OpenSatelliteStreamClient
+	conn        *grpc.ClientConn
+	streamId    string
+
+	recvChan           chan []byte
+	recvLoopClosedChan chan struct{}
+
+	state uint32
+}
+
+// NewSatelliteStream opens a stream to a satellite over the StellarStation API.
+func NewSatelliteStream(satelliteId string, recvChan chan []byte) (SatelliteStream, error) {
+	s := &satelliteStream{
+		satelliteId:        satelliteId,
+		streamId:           "",
+		recvChan:           recvChan,
+		state:              OPEN,
+		recvLoopClosedChan: make(chan struct{}),
+	}
+
+	return s, nil
+}
+
+// Start starts listening for data from the satellite.
+func (ss *satelliteStream) Start() error {
```

Comment:
Same here as the UDP proxy?

Reply:
Yeah actually it simplified some things too.